pkg/cli/policy_server.go

```diff
@@ -88,20 +88,20 @@ func (c *PolicyServerCommand) RunUnstarted(ctx context.Context, args []string) (
 	}
 	logger.Debugw("loaded configuration", "config", c.cfg)
 
-	successMessenger, err := server.NewPubSubMessenger(ctx, c.cfg.ProjectID, c.cfg.SuccessTopicID)
+	successTopic, err := createMessangerTopic(ctx, c.cfg.ProjectID, c.cfg.SuccessTopicID)
 	if err != nil {
 		return nil, nil, closer, fmt.Errorf("failed to create success event messenger: %w", err)
 	}
 
+	successMessenger := server.NewPubSubMessenger(successTopic)
+
 	handler, err := server.NewHandler(ctx, []server.Processor[*structpb.Struct]{}, successMessenger)
 	if err != nil {
 		return nil, nil, closer, fmt.Errorf("server.NewHandler: %w", err)
 	}
 
 	closer = func() {
-		if err := handler.Cleanup(); err != nil {
-			logger.Errorw("failed to close clean up handler", "error", err)
-		}
+		successTopic.Stop()
```

Comment:
To your surprise, you don't have to - see https://pkg.go.dev/cloud.google.com/go/pubsub#Client.Close. But I don't mind we continue to close it since it's trivial to do that

> If the client is available for the lifetime of the program, then Close need not be called at exit.

Reply:
@sqin2019 and I discussed this, from this reference [https://stackoverflow.com/questions/58351084/does-go-automatically-close-resources-if-not-explicitly-closed], we think it's best to call Close() anywhere to prevent some magic from happening.

Reply:
I don't think the reference means anything... If you don't call a func, Go won't call the func. That what it says. But the pubsub doc says you don't have to call it.